app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -122,39 +177,65 @@ const selectedTag = computed(() => {
 const selectedLabel = computed(() => {
   return selectedTag.value?.label
 })
-const innerWidgetInput = computed(() => {
-  if (props.input.dynamicConfig == null) return props.input
-  const config = props.input.dynamicConfig
-  if (config.kind !== 'Single_Choice') return props.input
-  return { ...props.input, dynamicConfig: singleChoiceConfiguration(config) }
+const innerWidgetInput = computed<WidgetInput>(() => {
+  const dynamicConfig =
+    props.input.dynamicConfig?.kind === 'Single_Choice' ?
+      singleChoiceConfiguration(props.input.dynamicConfig)
```

Comment:
Before, we were passing down the dynamic config if it is *not* Single_Choice. Now we don’t. Is it deliberate?

Reply:
No. I restored the old way, although I think there is not much difference.